ts_gazelle/BUILD

```diff
@@ -0,0 +1,13 @@
+load("@aspect_rules_ts//ts:defs.bzl", "ts_project")
+load("@bazel_gazelle//:def.bzl", "gazelle")
+
+gazelle(
+    name = "gazelle",
+    data = [],
+    gazelle = "@aspect_rules_ts_gazelle//:gazelle_js_binary",
+)
+
+ts_project(
+    name = "ts_gazelle",
+    srcs = ["lib/foo.ts"],
```

Comment:
@jbedard I confirmed that from userland we get the same behavior as the gazelle_generation_test: we don't ever create new BUILD files

Reply:
That seems to be the case. As I understand it gazelle invokes the `GenerateRules` method based on which BUILDs already exist.

See the sub-directories logic which I essentially copied from python:
https://github.com/bazelbuild/rules_python/blob/0.8.1/gazelle/generate.go#L105
https://github.com/aspect-build/rules_ts/blob/2ae033f0e4ea0ec1081de67231f7c07347797d96/gazelle/generate.go#L192